sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -423,6 +423,7 @@ object FunctionRegistry {
     expression[MakeTimestamp]("make_timestamp"),
     expression[MakeInterval]("make_interval"),
     expression[DatePart]("date_part"),
+    expression[Extract]("extract"),
```

Comment:
one side effect is now we support `extract(field, source)` other than `extract(field from source)`. Not a big deal but better if we can avoid exposing more APIs,

Reply:
> Not a big deal but better if we can avoid exposing more APIs

Yea, +1 .

btw, its better to add tests for the case `extract(field, source)`?

Reply:
Yea if we decide to support it.

Reply:
ok, I'll do follow-up later.

Reply:
I opened https://github.com/apache/spark/pull/28276